src/core/public/plugins/plugin_context.ts

```diff
@@ -40,24 +40,37 @@ export interface PluginInitializerContext {
     mode: Readonly<EnvironmentMode>;
     packageInfo: Readonly<PackageInfo>;
   };
+  readonly config: {
+    create: <T = ConfigSchema>() => Observable<T>;
+  };
 }
 
 /**
  * Provides a plugin-specific context passed to the plugin's construtor. This is currently
  * empty but should provide static services in the future, such as config and logging.
  *
  * @param coreContext
- * @param pluginManinfest
+ * @param opaqueId
+ * @param pluginManifest
+ * @param pluginConfig
  * @internal
  */
 export function createPluginInitializerContext(
   coreContext: CoreContext,
   opaqueId: PluginOpaqueId,
-  pluginManifest: DiscoveredPlugin
+  pluginManifest: DiscoveredPlugin,
+  pluginConfig: {
+    [key: string]: unknown;
+  }
 ): PluginInitializerContext {
   return {
     opaqueId,
     env: coreContext.env,
+    config: {
+      create<T>() {
+        return of<T>((pluginConfig as unknown) as T);
```

Comment:
Are we going to support runtime config updates on the client-side? Client-side plugins have a shorter time of life comparing to server-side ones. Should we provide a sync getter interface to simplify consumers' logic?

Reply:
This is an excellent question.

There are two reasons for this:
- Somewhere in the initial ticket I think I read that as NP was going to be SPA, the actual client app reload frequency was going to drastically decrease, and that we *might* want to support client-side runtime updates at some point, so this futur-proof the API. 
- It's consistent with the server counterpart.

However currently, this returns an 'sync' `of()` observable, so a sync getter can easily be added or even replace the `#create` function. Drawback is that if at some point we actually implements client config reload, we'll have to remove this sync getter.

Not sure how realistic implementing the client config reload is in short/mid term, so I don't know what is best here.

Reply:
> Somewhere in the initial ticket I think I read that as NP was going to be SPA, the actual client app reload frequency was going to drastically decrease, and that we might want to support client-side runtime updates at some point, so this futur-proofs the API.

I'm not opposed to the idea.  Albeit I see it is as a reasonable compromise, that platform sacrifice this functionality for the sake of plugin API simplicity. The platform can support config updates requesting a user to reload a page. We already do this for `uiSettings`.
https://github.com/elastic/kibana/blob/master/src/core/server/ui_settings/types.ts#L102
It reminds us that not all the parameters can be updated run time.
The main use case for ui config is `myplugin.ui.enabled`. It's can be updated in runtime as long as plugins use it with stateless API, e.g. to render content conditionally in React. Things get complicated when it's used with stateful API, like register a section on the management page or application registration.

Also, we can provide a sync config getter now and expose config stream when and if needed.

Reply:
Ok, sounds good to me. Will change for sync getter.